File: 0053. Maximum Subarray/solution.c
// Kadene's Algorithm
int maxSubArray(int* nums, int numsSize){
    int max_sum = -1000000000;
    int current_sum = 0;

    for (int i = 0; i < numsSize; i++)
    {
       current_sum = (current_sum + nums[i] > nums[i]) ? current_sum + nums[i] : nums[i];
        max_sum = (max_sum > current_sum) ? max_sum : current_sum;
    }
    return max_sum;

}

File: 0053. Maximum Subarray/solution.py
# Kadene's Algorithm
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxsum = float("-inf")
        cur = float("-inf")

        for num in nums:
            cur = max(cur + num, num)
            maxsum = max(maxsum, cur)

        return maxsum